renderdoc/driver/vulkan/wrappers/vk_cmd_funcs.cpp

```diff
@@ -1258,6 +1258,9 @@ bool WrappedVulkan::Serialise_vkBeginCommandBuffer(SerialiserType &ser, VkComman
               BeginInfo.pInheritanceInfo->subpass;
           // framebuffer is not useful here since it may be incomplete (imageless) and it's
           // optional, so we should just treat it as never present.
+
+          m_BakedCmdBufferInfo[BakedCommandBuffer].state.dynamicRendering.localRead.Init(
+              (const VkBaseInStructure *)BeginInfo.pInheritanceInfo);
```

Comment:
Forking slightly from other conversations to expand a bit on what this is doing and what is needed for state tracking. You can skip this comment if you want, as it's just informational to explain what you were unsure about and nothing needs to change as the code looks right now.

---

The idea of baked command buffers is related to the command buffer lifecycle itself and versioning. Imagine a `command buffer 1234` which is created, in the normal API this is recorded, then submitted 1 to N times, then potentially reset and used again. While using this 'live' there's no ambiguity about what that `command buffer 1234` handle refers to, but when replaying a capture there can be ambiguity if the command buffer is reset an re-recorded multiple times across the scope of the capture.

To address this RenderDoc has the concept of 'baked' command buffers. Whenever you begin/end `command buffer 1234` that particular _version_ of it gets treated as if a new object was created `baked command buffer 1235` which is immutable and only used whenever as that set of recorded commands is. If there is a rerecord, that becomes `baked command buffer 1236`. You can see this in the event browser if you look as command buffers will be referred to as 'baked'. There are some more details around ensuring the right version is used at the right time but that's not needed to understand this.

Given that, the baked command buffer info is not "baked (command buffer info)" but rather "(baked command buffer) info". Each version - of which there is almost always just one anyway - gets its own info so that its state, submissions, events etc can all be tracked properly.

Regarding the state tracking, generally we don't need to track any state during the initial load because we're replaying the whole capture and we don't need to care about the state along the way for displaying in the UI or for doing partial-replays of command buffers. We _do_ end up tracking a subset of state to be able to construct the information about resource 'usage in frame' - namely descriptor set and framebuffer bindings so that we know which images are used for what at any given event. That's why the `BeginRendering` dynamic rendering functions will be doing some state tracking that you might have seen, but we don't need to track anything that doesn't affect which resources are used like dynamic depth state or in this case a dynamic permutation of attachments.

---

Hopefully that makes sense as an info dump, like I say this all looks good now.